### DISCUSSION TITLE
downloadBriefcase by change set or index.

### DISCUSSION THREAD

**Discussion by beolson** [iTwin/itwinjs-core#5862] [General] [answered]
I have an implementation calling BriefcaseManager.downloadBriefcase that is working, but I now need to specify the version of that briefcase.  

I see the "asOf" argument, but have some questions about it.  It looks like I can pass in a change set in the "afterChangeSetId", but the phrasing of that is a little odd - is this pulling down the change set I pass in, or is pulling in a change set after the change set I pass in?

In addition, I have a test model that consists of the "Coffs Harbor" sample, and there is no change set - of, I guess it does, it is just an empty string.  Not sure if that is because there is only one "change set" in this model.  It does have an "index" of 0.  But, I dont see any way to pass in index into the downloadBriefcase call.  

![image](https://github.com/iTwin/itwinjs-core/assets/359112/a48e5e20-ca79-416b-b294-421c88950811)

**Comment by nick4598** [comment, marked as answer]
Hi,
I agree with you that the 'afterChangeSetId' property is a bit confusing. Looking at the code, calling IModelVersion.asOfChangeSet(changesetId: string) will populate the 'afterChangeSetId' property, and that specific function has this as the comment describing it:

>/** Describes a version of the iModel by the last change set that needs
>   to be applied or merged to the iModel.
>   Note that all ChangeSets up to and and including the specified ChangeSet
>   needs to be applied.
>   If the changesetId is an empty string, it is assumed to be the first version
>   before any change sets have been applied.
>   */
  
Therefore the afterChangeSetId property will pull down the changeset that you pass in. I will add an item to the backlog to better describe the afterChangeSetId property. 
 
 For your second question, you can pass an empty string in as the changesetId to signify that you want the first version of the imodel before any change sets have been applied (which appears to be the only version of the imodel according to your description).
 
 Let me know if this doesn't answer your questions.

**Comment by beolson** [comment]
That helps.  It sounds like the only time changeset will be blank is if it is the first version.  Every other version after that will have a changeset on it, right?

**Reply by nick4598** [reply]
Yes thats correct. You could think of that empty changesetId (index 0) as the baseline or seed file of the iModel as well.
